batch/sc-batch-fargate.yaml

```diff
@@ -0,0 +1,246 @@
+        Type: FARGATE
+        MaxvCpus: 1
+        Subnets:
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet1
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet2
+        SecurityGroupIds:
+          - !Ref SecurityGroup
+  ServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service:
+                - 'batch.amazonaws.com'
+            Action:
+              - 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole'
+  JobQueue:
+    Type: AWS::Batch::JobQueue
+    Properties:
+      ComputeEnvironmentOrder:
+        - ComputeEnvironment: !Ref ComputeEnvironment
+          Order: 1
+      Priority: 1
+      State: ENABLED
+  SchedulingPolicy:
+    Type: AWS::Batch::SchedulingPolicy
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3001
+    Properties:
+      FairsharePolicy:
+        ShareDecaySeconds: 5
+        ComputeReservation: 2
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      VpcId: !ImportValue us-east-1-scriptrunnervpc-VPCId
+      GroupDescription: !Sub 'SC for ${AWS::StackName}}'
+  ScriptRunnerSecrets:
+    Type: 'AWS::SecretsManager::Secret'
+    Properties:
+      SecretString: !Sub '{ ${Secrets} }'
+  # Allow read access to ScriptRunnerSecrets
+  # https://docs.aws.amazon.com/mediaconnect/latest/ug/iam-policy-examples-asm-secrets.html
+  SecretsAccessPolicy:
+    Type: 'AWS::IAM::ManagedPolicy'
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - secretsmanager:GetResourcePolicy
+              - secretsmanager:GetSecretValue
+              - secretsmanager:DescribeSecret
+              - secretsmanager:ListSecretVersionIds
+            Resource:
+              - !Ref ScriptRunnerSecrets
+          - Effect: Allow
+            Action: secretsmanager:ListSecrets
+            Resource: '*'
+  JobDefinition:
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:  # Ignore cfn-lint errors for PyPlate.
+            - E3002
+    Type: AWS::Batch::JobDefinition
+    Properties:
+      Type: container
+      JobDefinitionName: { Ref: "AWS::StackName" }
+      PropagateTags: True
+      PlatformCapabilities:
+        - FARGATE
+      Timeout:
+        AttemptDurationSeconds: 600
+      RetryStrategy:
+        Attempts: 1
+      ContainerProperties:
+        Command: !Ref Command
+        Image: !Ref Image
+        Environment: |
+          #!PyPlate
+          output = []
+          for tag in params['EnvVars']:
+             key, value = tag.split('=')
+             output.append({"Name": key, "Value": value})
+        Secrets:
```

Comment:
I like this design a lot... And I think it would be most straightforward to put all env vars in secretsmanager and to remove the `Environment` parameter to `ContainerProperties`. This would remove the dependency on PyPlate parsing and would require end users to write container jobs that interpret environment inputs in the way we want them to.

Reply:
I'm not a fan of this suggestion for the following reasons:
1.  There is a cost when adding vars to secrets manager which means it would be more costly
2. Putting vars that are not secrets into the secrets manager would make it more difficult to debug issues.
3. Using it that way is not how batch was designed.  It just feels like we would be swimming upstream to how batch should be used.